frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts

```diff
@@ -25,25 +25,42 @@ type UseFetchKeysAndValuesReturnValues = {
 export const useFetchKeysAndValues = (
 	searchValue: string,
 	query: IBuilderQueryForm,
-): UseFetchKeysAndValuesReturnValues => {
-	const [keys, setKeys] = useState<AttributeKeyOptions[]>([]);
+): IuseFetchKeysAndValues => {
+	const [keys, setKeys] = useState<BaseAutocompleteData[]>([]);
 	const [results, setResults] = useState<string[]>([]);
+
+	const isQueryEnabled = useMemo(
```

Comment:
@techchintan no need of memo as return value is primitive

Reply:
This is not a primitive value case, the left side of the condition is a non-primitive  `query.dataSource`
So based on your `query.dataSource` it will change boolean.